### ISSUE TITLE
Matrix4 entries must be finite

### ISSUE THREAD

**Issue by kirill-21** [gskinner/flutter_animate#40]
```dart
'dart:ui/painting.dart': Failed assertion: line 50 pos 10: '<optimized out>': Matrix4 entries must be finite., #0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      _matrix4IsValid (dart:ui/painting.dart:50:10)
#3      new Gradient.linear (dart:ui/painting.dart:3726:34)
#4      LinearGradient.createShader (package:flutter/src/painting/gradient.dart:436:24)
#5      ShimmerEffect.build.<anonymous closure>.<anonymous closure> (package:flutter_animate/effects/shimmer_effect.dart:70:48)
#6      RenderShaderMask.paint (package:flutter/src/rendering/proxy_box.dart:1188:35)
#7      RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#8      PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#9      PaintingContext.pushLayer (package:flutter/src/rendering/object.dart:460:12)
#10     PaintingContext.pushClipPath (package:flutter/src/rendering/object.dart:600:7)
#11     RenderClipPath.paint (package:flutter/src/rendering/proxy_box.dart:1830:25)
#12     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#13     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#14     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#15     RenderDecoratedBox.paint (package:flutter/src/rendering/proxy_box.dart:2371:11)
#16     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#17     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#18     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#19     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#20     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#21     RenderBoxContainerDefaultsMixin.defaultPaint (package:flutter/src/rendering/box.dart:2900:15)
#22     RenderStack.paintStack (package:flutter/src/rendering/stack.dart:654:5)
#23     RenderStack.paint (package:flutter/src/rendering/stack.dart:670:7)
#24     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#25     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#26     RenderShiftedBox.paint (package:flutter/src/rendering/shifted_box.dart:84:15)
#27     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#28     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#29     PaintingContext.pushLayer (package:flutter/src/rendering/object.dart:460:12)
#30     PaintingContext.pushOpacity (package:flutter/src/rendering/object.dart:693:5)
#31     RenderOpacity.paint (package:flutter/src/rendering/proxy_box.dart:954:21)
#32     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#33     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#34     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#35     RenderTransform.paint (package:flutter/src/rendering/proxy_box.dart:2617:17)
#36     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#37     PaintingContext.paintChild (package:flutter/src/rendering/object.dart:239:13)
#38     RenderProxyBoxMixin.paint (package:flutter/src/rendering/proxy_box.dart:144:15)
#39     RenderObject._paintWithContext (package:flutter/src/rendering/object.dart:2796:7)
#40     PaintingContext._repaintCompositedChild (package:flutter/src/rendering/object.dart:155:11)
#41     PaintingContext.repaintCompositedChild (package:flutter/src/rendering/object.dart:98:5)
#42     PipelineOwner.flushPaint (package:flutter/src/rendering/object.dart:1116:31)
#43     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:515:19)
#44     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:884:13)
#45     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:378:5)
#46     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1175:15)
#47     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1104:9)
#48     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:1015:5)
#49     _invoke (dart:ui/hooks.dart:148:13)
#50     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:318:5)#51     _drawFrame (dart:ui/hooks.dart:115:31)
```

Unfortunately i can not provide reproducible steps because it's very hard to recreate this issue in a new project. It happens on Windows when i minimize app with custom PostMessage method and use ShimmerEfect on a GridView child. Windows makes app smaller to show hiding animation and this issue appears.

![image](https://user-images.githubusercontent.com/63228361/203137868-c523bba9-2478-48d1-a481-0ef0dd87c55a.png)
It seems that width of gridview's child become 0 after minimizing the app.

Can something like this be implemented as a fix or do not play animation at all if one of child's constraints is 0?
![image](https://user-images.githubusercontent.com/63228361/203139242-3ef46b59-ead0-4ac6-b0cb-48251260cc03.png)

To get the same error output you can just create a container with 0 width and apply this effect to it
```dart
Animate(
            onPlay: (controller) => controller.repeat(reverse: true),
            effects: [
              ShimmerEffect(
                color: Colors.white.withOpacity(0.40),
                size: 2,
                blendMode: BlendMode.srcATop,
                delay: const Duration(milliseconds: 350),
                duration: const Duration(milliseconds: 1000),
              ),
            ],
            child: Container(
              color: Colors.black,
              width: 0,
              height: 80,
              child: Text('Hello World'),
            ),
          )
```

**Comment by gskinner** [comment]
At a quick glance, I _think_ the best way to fix this would be in `_SweepingGradientTransform.transform` by checking if either dimension of `bounds` is `0`, and setting it to `0.1` (or whatever works).

I'll try to take a look at this soon, but feel free to try it out and report back.

**Comment by gskinner** [comment]
(fix released in v2.1.0)
